In [3]:
import pandas as pd
import numpy as np
import os
os.getcwd()

'C:\\Users\\Ankita\\OneDrive - Infiniti Research\\Desktop\\Johnson&Johnson\\Data'

# Reading datasets

In [4]:
sales=pd.read_excel(r'C:/Users/Ankita/OneDrive - Infiniti Research/Desktop/Johnson&Johnson/Data/Sales_Ratings_Reviews_neutro.xlsx', sheet_name='Sales_Data')
sales

year   date  weeknum  fiscal_year  fiscal_month   retailer  country  \
0      2021  44470       39         2021             9  Amazon 1P  Germany   
1      2021  44470       39         2021             9  Amazon 1P  Germany   
2      2021  44470       39         2021             9  Amazon 1P  Germany   
3      2021  44470       39         2021             9  Amazon 1P  Germany   
4      2021  44470       39         2021             9  Amazon 1P  Germany   
...     ...    ...      ...          ...           ...        ...      ...   
14351  2022  44624        9         2022             3  Amazon 1P  Germany   
14352  2022  44624        9         2022             3  Amazon 1P  Germany   
14353  2022  44624        9         2022             3  Amazon 1P  Germany   
14354  2022  44624        9         2022             3  Amazon 1P  Germany   
14355  2022  44624        9         2022             3  Amazon 1P  Germany   

            manufacturer       category       brand  ...  \
0      JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
1      JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
2      JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
3      JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
4      JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
...                  ...            ...         ...  ...   
14351  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
14352  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
14353  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
14354  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
14355  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   

      prep_instructions_required_vc prep_instructions_vendor_state_vc  \
0                                 -                            Amazon   
1                                 -                            Vendor   
2             Set creation, Bagging                            Amazon   
3                                 -                            Amazon   
4             Bagging, Set creation                            Amazon   
...                             ...                               ...   
14351                             -                            Amazon   
14352                             -                            Amazon   
14353                             -                    CPEX Certified   
14354                             -                                 -   
14355                           NaN                               NaN   

      brand_code_vc    brand_vc manufacturer_code_vc  \
0             NEUVP  Neutrogena                JOMO1   
1             NEUVP  Neutrogena                JOMO1   
2             NEUVP  Neutrogena                JOMO1   
3             NEUVP  Neutrogena                JOMO1   
4             NEUVP  Neutrogena                JOMO1   
...             ...         ...                  ...   
14351         NEUVP  Neutrogena                JOMO1   
14352         NEUVP  Neutrogena                JOMO1   
14353         NEUXP  Neutrogena                JOMO1   
14354         NEUXP  Neutrogena                JOMO1   
14355           NaN         NaN                  NaN   

      parent_manufacturer_code_vc local_currency  target_currency   ex_rate  \
0                           JOMO1            EUR              USD  1.198843   
1                           JOMO1            EUR              USD  1.198843   
2                           JOMO1            EUR              USD  1.198843   
3                           JOMO1            EUR              USD  1.198843   
4                           JOMO1            EUR              USD  1.198843   
...                           ...            ...              ...       ...   
14351                       JOMO1            EUR              USD  1.198843   
14352                       JOMO1            EUR              USD  1.198843   
14353                       JOMO1            EUR              USD  1.198843   
14354    

In [5]:
df_reviews=pd.read_excel(r'C:/Users/Ankita/OneDrive - Infiniti Research/Desktop/Johnson&Johnson/Data/Sales_Ratings_Reviews_neutro.xlsx', sheet_name='Reviews')
df_reviews 

year   date  weeknum  fiscal_year  fiscal_month   retailer  country  \
0     2022  44656       14         2022             4  Amazon 1P  Germany   
1     2022  44664       15         2022             4  Amazon 1P  Germany   
2     2022  44675       17         2022             4  Amazon 1P  Germany   
3     2022  44718       23         2022             6  Amazon 1P  Germany   
4     2022  44728       24         2022             6  Amazon 1P  Germany   
...    ...    ...      ...          ...           ...        ...      ...   
5338  2022  44649       13         2022             3  Amazon 1P  Germany   
5339  2022  44650       13         2022             3  Amazon 1P  Germany   
5340  2022  44658       14         2022             4  Amazon 1P  Germany   
5341  2022  44672       16         2022             4  Amazon 1P  Germany   
5342  2022  44673       16         2022             4  Amazon 1P  Germany   

           manufacturer       category       brand  ... unique_pr  \
0     JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...         1   
1     JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...         1   
2     JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...         1   
3     JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...         1   
4     JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...         1   
...                 ...            ...         ...  ...       ...   
5338  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...         0   
5339  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...         0   
5340  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...         0   
5341  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...         0   
5342  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...         0   

                    duplicate_group_pr  \
0     ac233f7e2cb51e79bcd756a6ed27511e   
1     2148a42f918d60a38c24066d9e46d2c6   
2     815d756a15f03c2adffed505948d80a2   
3     812c16799d89c56b11b5e08902565985   
4     2997449e348b2a4bf963e82c8873f2f8   
...                                ...   
5338  cbfeb1eb820c029f4351e8af73816833   
5339  925e9103194684c9d065d845b3f4acc2   
5340  28b2da8b002f1d6f746f9af2a8740172   
5341  b951d3b4e3fc8d6c7a743b9f78e466e1   
5342  37f79aa7ebbd308a5cfdb0fbdb514d51   

                                          review_url_pr  \
0     https://www.amazon.de/gp/customer-reviews/R3OG...   
1     https://www.amazon.de/gp/customer-reviews/R1VE...   
2     https://www.amazon.de/gp/customer-reviews/R2UP...   
3     https://www.amazon.de/gp/customer-reviews/R1E7...   
4     https://www.amazon.de/gp/customer-reviews/R2O4...   
...                                                 ...   
5338  https://www.amazon.de/gp/customer-reviews/RBN7...   
5339  https://www.amazon.de/gp/customer-reviews/RDX3...   
5340  https://www.amazon.de/gp/customer-reviews/RT7M...   
5341  https://www.amazon.de/gp/customer-reviews/R36A...   
5342  https://www.amazon.de/gp/customer-reviews/R18V...   

     manufacturer_comment_pr manufacturer_comment_text_pr  \
0                          0                          NaN   
1                          0                          NaN   
2                          0                          NaN   
3                          0                          NaN   
4                          0                          NaN   
...                      ...                          ...   
5338                       0                          NaN   
5339                       0                          NaN   
5340                       0                          NaN   
5341                       0                          NaN   
5342                       0                          NaN   

     manufacturer_comment_date_pr has_image_pr  \
0                             NaN            0   
1                             NaN            0   
2                             NaN            0   
3                             NaN            0   
4                             NaN            0   
...  

In [6]:
df_ratings=pd.read_excel(r'C:/Users/Ankita/OneDrive - Infiniti Research/Desktop/Johnson&Johnson/Data/Sales_Ratings_Reviews_neutro.xlsx', sheet_name='Ratings')
df_ratings   

year   date  weeknum  fiscal_year  fiscal_month   retailer  country  \
0     2021  44358       23         2021             6  Amazon 1P  Germany   
1     2021  44360       24         2021             6  Amazon 1P  Germany   
2     2021  44361       24         2021             6  Amazon 1P  Germany   
3     2021  44363       24         2021             6  Amazon 1P  Germany   
4     2021  44364       24         2021             6  Amazon 1P  Germany   
...    ...    ...      ...          ...           ...        ...      ...   
9193  2022  44745       27         2022             6  Amazon 1P  Germany   
9194  2022  44746       27         2022             7  Amazon 1P  Germany   
9195  2022  44749       27         2022             7  Amazon 1P  Germany   
9196  2022  44753       28         2022             7  Amazon 1P  Germany   
9197  2022  44762       29         2022             7  Amazon 1P  Germany   

           manufacturer       category       brand  ... regular_price_pr  \
0     JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...             4.55   
1     JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...             4.55   
2     JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...             4.55   
3     JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...             4.55   
4     JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...             4.55   
...                 ...            ...         ...  ...              ...   
9193  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...             9.94   
9194  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...             9.94   
9195  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...             9.94   
9196  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...             7.99   
9197  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...             9.49   

     promotion_text_pr promotion_price_pr availability_pr  \
0                  NaN                NaN               Y   
1                  NaN                NaN               Y   
2                  NaN                NaN               Y   
3                  NaN                NaN               Y   
4                  NaN                NaN               Y   
...                ...                ...             ...   
9193               NaN                NaN               Y   
9194               NaN                NaN               Y   
9195               NaN                NaN               Y   
9196               NaN                NaN               Y   
9197               NaN                NaN               Y   

     first_party_won_buy_box_pr third_party_seller_pr add_on_item_pr  \
0                           1.0                amazon              0   
1                           1.0                amazon              0   
2                           1.0                amazon              0   
3                           1.0                amazon              0   
4                           1.0                amazon              0   
...                         ...                   ...            ...   
9193                        1.0                amazon              0   
9194                        1.0                amazon              0   
9195                        1.0                amazon              0   
9196                        1.0                amazon              0   
9197                        1.0                amazon              0   

      prime_exclusive_pr promo_type_pr         date&asin  
0                    0.0           NaN  44358_B000W6PY1W  
1                    0.0           NaN  44360_B000W6PY1W  
2                    0.0           NaN  44361_B000W6PY1W  
3                    0.0           NaN  44363_B000W6PY1W  
4                    0.0           NaN  44364_B000W6PY1W  
...                  ...           ...               ...  
9193                 NaN           NaN  44745_B09GYRXP9P  
9194                 NaN           NaN  44746_B09GYRXP9P  
9195                 NaN           NaN 

In [7]:
df_placement=pd.read_excel(r'C:/Users/Ankita/OneDrive - Infiniti Research/Desktop/Johnson&Johnson/Data/Placement_neutro.xlsx', sheet_name='filtered_data')
df_placement  

year   date  weeknum  fiscal_year  fiscal_month   retailer  country  \
0       2021  44481       41         2021            10  Amazon 1P  Germany   
1       2021  44481       41         2021            10  Amazon 1P  Germany   
2       2021  44481       41         2021            10  Amazon 1P  Germany   
3       2021  44481       41         2021            10  Amazon 1P  Germany   
4       2021  44481       41         2021            10  Amazon 1P  Germany   
...      ...    ...      ...          ...           ...        ...      ...   
167237  2022  44620        9         2022             3  Amazon 1P  Germany   
167238  2022  44620        9         2022             3  Amazon 1P  Germany   
167239  2022  44620        9         2022             3  Amazon 1P  Germany   
167240  2022  44620        9         2022             3  Amazon 1P  Germany   
167241  2022  44621        9         2022             3  Amazon 1P  Germany   

             manufacturer       category       brand  ... prime_pantry_pr  \
0       JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...           False   
1       JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...           False   
2       JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...           False   
3       JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...           False   
4       JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...           False   
...                   ...            ...         ...  ...             ...   
167237  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...           False   
167238  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...           False   
167239  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...           False   
167240  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...           False   
167241  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...           False   

       amazon_subscribe_and_save_pr amazon_prime_pr amazon_addon_item_pr  \
0                             False           False                False   
1                              True            True                 True   
2                             False           False                False   
3                             False           False                False   
4                             False           False                False   
...                             ...             ...                  ...   
167237                        False           False                False   
167238                        False           False                False   
167239                        False           False                False   
167240                        False           False                False   
167241                        False           False                False   

       star_rating_pr review_count_pr number_of_images_pr        type_pr  \
0                 4.6            62.0                   8  brand_keyword   
1                 4.6            62.0                   8  brand_keyword   
2                 4.6            62.0                   8  brand_keyword   
3                 4.6            62.0                   8  brand_keyword   
4                 4.6            62.0                   8  brand_keyword   
...               ...             ...                 ...            ...   
167237            4.5            96.0                   6  brand_keyword   
167238            4.5            96.0                   6  brand_keyword   
167239            4.5            96.0                   6  brand_keyword   
167240            4.5            96.0                   6  brand_keyword   
167241            4.5            96.0                   6  brand_keyword   

               date&asin                                      url_pr  
0       44481_B08HT315T6  https://www.amazon.de/dp/B08HT315T6/?psc=1  
1       44481_B08HT315T6  https://www.amazon.de/dp/B08HT315T6/?psc=1  
2       44481_B08HT315T6  https://www.amazon.de/dp/B08HT315T6/?psc=1  
3       44481_B08HT315T6

In [8]:
df_content=pd.read_excel(r'C:/Users/Ankita/OneDrive - Infiniti Research/Desktop/Johnson&Johnson/Data/Content_neutro.xlsx', sheet_name='Sheet1')
df_content  

year   date  weeknum  fiscal_year  fiscal_month   retailer  country  \
0      2022  44579        3         2022             1  Amazon 1P  Germany   
1      2022  44579        3         2022             1  Amazon 1P  Germany   
2      2021  44349       22         2021             6  Amazon 1P  Germany   
3      2021  44282       12         2021             3  Amazon 1P  Germany   
4      2021  44282       12         2021             3  Amazon 1P  Germany   
...     ...    ...      ...          ...           ...        ...      ...   
83441  2022  44632       10         2022             3  Amazon 1P  Germany   
83442  2022  44632       10         2022             3  Amazon 1P  Germany   
83443  2022  44632       10         2022             3  Amazon 1P  Germany   
83444  2022  44632       10         2022             3  Amazon 1P  Germany   
83445  2022  44632       10         2022             3  Amazon 1P  Germany   

            manufacturer       category       brand  ... gsr_level_4  \
0      JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...         NaN   
1      JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...         NaN   
2      JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...         NaN   
3      JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...         NaN   
4      JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...         NaN   
...                  ...            ...         ...  ...         ...   
83441  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...      Others   
83442  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...      Others   
83443  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...      Others   
83444  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...      Others   
83445  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...      Others   

      is_competitor                     content_field_pr condition_pr  \
0             False                  A+ Content (Exists)            Y   
1             False                   A+ Images (Exists)            Y   
2             False         Number of Images (Benchmark)            Y   
3             False                  A+ Content (Exists)          NaN   
4             False                  A+ Content (Exists)          NaN   
...             ...                                  ...          ...   
83441         False  Length of Product Title (Benchmark)            N   
83442         False         Number of Images (Benchmark)            Y   
83443         False             Number of Images (Count)            Y   
83444         False      Product Description (Benchmark)            Y   
83445         False           Product Description Exists            Y   

      compliance_manually_changed_pr      expected_content_pr  \
0                                0.0                   Exists   
1                                0.0                        1   
2                                0.0                   Exists   
3                                NaN                      NaN   
4                                0.0                      NaN   
...                              ...                      ...   
83441                            0.0  30 ≤ x ≤ 150 Characters   
83442                            0.0                        6   
83443                            0.0                   Exists   
83444                            0.0                     =>80   
83445                            0.0                   Exists   

                                       actual_content_pr  critical_pr  \
0                                                    Yes        False   
1                                                      5        False   
2                                                     11        False   
3                                                    Yes        False   
4                                                    Yes        False   
...                                                  ...          ...   
83441  194: Neutrogena Anti-Pickel+ 

# Dropping the concatenated column

In [9]:
df_reviews=df_reviews.drop('date&asin',axis=1)

In [10]:
df_ratings=df_ratings.drop('date&asin',axis=1)
df_placement=df_placement.drop('date&asin',axis=1)
df_content=df_content.drop('date&asin',axis=1)

# Merging datasets

In [11]:
df_final=pd.merge(sales,df_reviews,on=['date','asin'])
df_final

year_x   date  weeknum_x  fiscal_year_x  fiscal_month_x retailer_x  \
0       2021  44304         16           2021               4  Amazon 1P   
1       2021  44304         16           2021               4  Amazon 1P   
2       2021  44304         16           2021               4  Amazon 1P   
3       2021  44304         16           2021               4  Amazon 1P   
4       2021  44304         16           2021               4  Amazon 1P   
...      ...    ...        ...            ...             ...        ...   
4580    2022  44624          9           2022               3  Amazon 1P   
4581    2022  44624          9           2022               3  Amazon 1P   
4582    2022  44624          9           2022               3  Amazon 1P   
4583    2022  44624          9           2022               3  Amazon 1P   
4584    2022  44624          9           2022               3  Amazon 1P   

     country_x     manufacturer_x     category_x     brand_x  ...  \
0      Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
1      Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
2      Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
3      Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
4      Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
...        ...                ...            ...         ...  ...   
4580   Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
4581   Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
4582   Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
4583   Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
4584   Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   

     star_rating_pr unique_pr                duplicate_group_pr  \
0                 5         1  6bab02b3da40d972a5f7bdafea1e5b36   
1                 5         1  5a3d4e57cac0ee5b25bad4278a613fa8   
2                 3         1  af8043916a19acf7a7dd31a0035abbb1   
3                 4         1  38bf347aebe244be64901a6e03dd4dc0   
4                 4         1  4173b9b75b65790a1be5279f7f89bbae   
...             ...       ...                               ...   
4580              1         1  27b6bcd7ff01068de9669d9a350641e2   
4581              5         1  cea6f9d458df47bb5108b3e5a91e0060   
4582              5         1  abf0cbed99dd894e8aa9dfedc6e7388a   
4583              5         1  6c7edcf3701ac12bae7ece54b55d871c   
4584              2         1  dff756905e68f885a8c512fb8a1e4d35   

                                          review_url_pr  \
0     https://www.amazon.de/gp/customer-reviews/R3FJ...   
1     https://www.amazon.de/gp/customer-reviews/R3SV...   
2     https://www.amazon.de/gp/customer-reviews/RRR1...   
3     https://www.amazon.de/gp/customer-reviews/RFR7...   
4     https://www.amazon.de/gp/customer-reviews/R3C1...   
...                                                 ...   
4580  https://www.amazon.de/gp/customer-reviews/R19D...   
4581  https://www.amazon.de/gp/customer-reviews/R3HT...   
4582  https://www.amazon.de/gp/customer-reviews/R3HT...   
4583  https://www.amazon.de/gp/customer-reviews/R1IE...   
4584  https://www.amazon.de/gp/customer-reviews/R1CG...   

     manufacturer_comment_pr manufacturer_comment_text_pr  \
0                          0                          NaN   
1                          0                          NaN   
2                          0                          NaN   
3                          0                          NaN   
4                          0                          NaN   
...                      ...                          ...   
4580                       0                          NaN   
4581                       0                          NaN   
4582                       0                          NaN   
4583                       0                          NaN   
4584                       0                          NaN   

     manufacturer

In [12]:
df_final=pd.merge(df_final,df_ratings,on=['date','asin'])
df_final

year_x   date  weeknum_x  fiscal_year_x  fiscal_month_x retailer_x  \
0       2021  44304         16           2021               4  Amazon 1P   
1       2021  44304         16           2021               4  Amazon 1P   
2       2021  44304         16           2021               4  Amazon 1P   
3       2021  44304         16           2021               4  Amazon 1P   
4       2021  44304         16           2021               4  Amazon 1P   
...      ...    ...        ...            ...             ...        ...   
3014    2022  44617          8           2022               2  Amazon 1P   
3015    2022  44617          8           2022               2  Amazon 1P   
3016    2022  44624          9           2022               3  Amazon 1P   
3017    2022  44624          9           2022               3  Amazon 1P   
3018    2022  44624          9           2022               3  Amazon 1P   

     country_x     manufacturer_x     category_x     brand_x  ...  \
0      Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
1      Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
2      Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
3      Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
4      Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
...        ...                ...            ...         ...  ...   
3014   Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
3015   Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
3016   Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
3017   Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
3018   Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   

                                          url_pr regular_price_pr_y  \
0     https://www.amazon.de/dp/B07BVW522M/?psc=1              10.95   
1     https://www.amazon.de/dp/B07BVW522M/?psc=1              10.95   
2     https://www.amazon.de/dp/B07BVW522M/?psc=1              10.95   
3     https://www.amazon.de/dp/B07BVW522M/?psc=1              10.95   
4     https://www.amazon.de/dp/B07BVW522M/?psc=1              10.95   
...                                          ...                ...   
3014  https://www.amazon.de/dp/B087ZGGRL5/?psc=1                NaN   
3015  https://www.amazon.de/dp/B087ZGGRL5/?psc=1                NaN   
3016  https://www.amazon.de/dp/B0849D9GB4/?psc=1               9.49   
3017  https://www.amazon.de/dp/B0849D9GB4/?psc=1               9.49   
3018  https://www.amazon.de/dp/B0849D9GB4/?psc=1               9.49   

                  promotion_text_pr_y promotion_price_pr_y availability_pr_y  \
0                                 NaN                  NaN                 Y   
1                                 NaN                  NaN                 Y   
2                                 NaN                  NaN                 Y   
3                                 NaN                  NaN                 Y   
4                                 NaN                  NaN                 Y   
...                               ...                  ...               ...   
3014  Sparen Sie 5 % beim Kauf von 4.                 4.89                 Y   
3015  Sparen Sie 5 % beim Kauf von 4.                 4.89                 Y   
3016                              NaN                  NaN                 Y   
3017                              NaN                  NaN                 Y   
3018                              NaN                  NaN                 Y   

     first_party_won_buy_box_pr_y third_party_seller_pr_y  add_on_item_pr_y  \
0                             NaN                     NaN                 0   
1                             NaN                     NaN                 0   
2                             NaN                     NaN                 0   
3                             NaN                     NaN                 0   
4                             NaN                     N

In [13]:
df_final=pd.merge(df_final,df_placement,on=['date','asin'])
df_final

C:\Users\Ankita\AppData\Local\Temp/ipykernel_8908/970916127.py:1: FutureWarning: Passing 'suffixes' which cause duplicate columns {'country_x', 'manufacturer_x', 'gfo_x', 'gsr_level_1_x', 'weeknum_x', 'category_x', 'gsr_level_2_x', 'gsr_level_3_x', 'year_x', 'title_x', 'is_competitor_x', 'fiscal_year_x', 'gsr_level_4_x', 'retailer_x', 'fiscal_month_x', 'brand_x'} in the result is deprecated and will raise a MergeError in a future version.
  df_final=pd.merge(df_final,df_placement,on=['date','asin'])


year_x   date  weeknum_x  fiscal_year_x  fiscal_month_x retailer_x  \
0        2021  44304         16           2021               4  Amazon 1P   
1        2021  44304         16           2021               4  Amazon 1P   
2        2021  44304         16           2021               4  Amazon 1P   
3        2021  44304         16           2021               4  Amazon 1P   
4        2021  44304         16           2021               4  Amazon 1P   
...       ...    ...        ...            ...             ...        ...   
19258    2022  44624          9           2022               3  Amazon 1P   
19259    2022  44624          9           2022               3  Amazon 1P   
19260    2022  44624          9           2022               3  Amazon 1P   
19261    2022  44624          9           2022               3  Amazon 1P   
19262    2022  44624          9           2022               3  Amazon 1P   

      country_x     manufacturer_x     category_x     brand_x  ...  \
0       Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
1       Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
2       Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
3       Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
4       Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
...         ...                ...            ...         ...  ...   
19258   Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
19259   Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
19260   Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
19261   Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
19262   Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   

      amazon_bestseller_pr prime_pantry_pr amazon_subscribe_and_save_pr  \
0                    False           False                        False   
1                    False           False                        False   
2                    False           False                        False   
3                    False           False                        False   
4                    False           False                        False   
...                    ...             ...                          ...   
19258                False           False                        False   
19259                 True            True                         True   
19260                 True            True                         True   
19261                False           False                        False   
19262                False           False                        False   

      amazon_prime_pr amazon_addon_item_pr star_rating_pr_y review_count_pr  \
0               False                False              4.5           120.0   
1               False                False              4.5           120.0   
2               False                False              4.5           120.0   
3               False                False              4.5           120.0   
4               False                False              4.5           120.0   
...               ...                  ...              ...             ...   
19258           False                False              4.6           154.0   
19259            True                 True              4.6           154.0   
19260            True                 True              4.6           154.0   
19261           False                False              4.6           154.0   
19262           False                False              4.6           154.0   

       number_of_images_pr        type_pr  \
0                        7  brand_keyword   
1                        7  brand_keyword   
2                        7  brand_keyword   
3                        7  brand_keyword   
4                        7  brand_keyword   
...                    ...            ...   
19258                    6  brand_keyword   
19259                    6  bra

In [14]:
df_final=pd.merge(df_final,df_content,on=['date','asin'])

In [15]:
df_final

year_x   date  weeknum_x  fiscal_year_x  fiscal_month_x retailer_x  \
0        2021  44304         16           2021               4  Amazon 1P   
1        2021  44304         16           2021               4  Amazon 1P   
2        2021  44304         16           2021               4  Amazon 1P   
3        2021  44304         16           2021               4  Amazon 1P   
4        2021  44304         16           2021               4  Amazon 1P   
...       ...    ...        ...            ...             ...        ...   
99219    2022  44624          9           2022               3  Amazon 1P   
99220    2022  44624          9           2022               3  Amazon 1P   
99221    2022  44624          9           2022               3  Amazon 1P   
99222    2022  44624          9           2022               3  Amazon 1P   
99223    2022  44624          9           2022               3  Amazon 1P   

      country_x     manufacturer_x     category_x     brand_x  ...  \
0       Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
1       Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
2       Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
3       Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
4       Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
...         ...                ...            ...         ...  ...   
99219   Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
99220   Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
99221   Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
99222   Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   
99223   Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...   

      gsr_level_3 gsr_level_4 is_competitor  \
0         Regular         NaN         False   
1         Regular         NaN         False   
2         Regular         NaN         False   
3         Regular         NaN         False   
4         Regular         NaN         False   
...           ...         ...           ...   
99219     Regular         NaN         False   
99220     Regular         NaN         False   
99221     Regular         NaN         False   
99222     Regular         NaN         False   
99223     Regular         NaN         False   

                          content_field_pr condition_pr  \
0                      A+ Content (Exists)            Y   
1                       A+ Images (Exists)            Y   
2                      A+ Content (Exists)            Y   
3                       A+ Images (Exists)            Y   
4                      A+ Content (Exists)            Y   
...                                    ...          ...   
99219  Length of Product Title (Benchmark)            N   
99220         Number of Images (Benchmark)            Y   
99221             Number of Images (Count)            Y   
99222      Product Description (Benchmark)            Y   
99223           Product Description Exists            Y   

      compliance_manually_changed_pr      expected_content_pr  \
0                                0.0                   Exists   
1                                0.0                        1   
2                                0.0                   Exists   
3                                0.0                        1   
4                                0.0                   Exists   
...                              ...                      ...   
99219                            0.0  30 ≤ x ≤ 150 Characters   
99220                            0.0                        6   
99221                            0.0                   Exists   
99222                            0.0                     =>80   
99223                            0.0                   Exists   

                                       actual_content_pr critical_pr  \
0                                                    Yes       False   
1                                         

In [16]:
df_final.shape

(99224, 229)

In [ ]:
import datetime as dt

In [115]:
df_final["date"] = pd.to_datetime(df_final["date"])

# Reading Datasets for uncommon columns in the filtered dataset

In [17]:
df_reviews1=pd.read_excel(r'C:/Users/Ankita/OneDrive - Infiniti Research/Desktop/Johnson&Johnson/Data/Sales_Ratings_Reviews_neutro.xlsx', sheet_name='unique_col_reviews')
df_reviews1

date        asin                summary_pr  \
0     44656  B082VMNCNV  Salicylsäure Maske Top !   
1     44664  B082VMNCNV         Does what it says   
2     44675  B082VMNCNV     Fazit nach einem Jahr   
3     44718  B082VMNCNV     Macht die Haut reiner   
4     44728  B082VMNCNV                       Gut   
...     ...         ...                       ...   
5338  44649  B094GC1G7N     Leichte Gesichtscreme   
5339  44650  B094GC1G7N             Einfach super   
5340  44658  B094GC1G7N                       Gut   
5341  44672  B094GC1G7N      Ich bin zufrieden...   
5342  44673  B094GC1G7N                Super soft   

                                                text_pr  star_rating_pr  \
0     Salicylsäure-Maske Top ! Muss sagen es glättet...               5   
1     It's not so hard on the skin but does work. sk...               5   
2     Nun nach der Zeit kann ich sagen, dass die Pro...               2   
3     Ist einfach anzuwenden und das Hautbild wird t...               4   
4                                                   Gut               5   
...                                                 ...             ...   
5338  Eine super leichte , feuchtigkeitsreiche Creme...               5   
5339  Hatte sie immer wieder gekauft in den letzten ...               5   
5340                                   Hat mir gefallen               5   
5341  Ich nutze die Tagescreme seit Jahren. Habe jet...               5   
5342  Es fühlt sich einfach geil an. Man braucht nur...               5   

      unique_pr                duplicate_group_pr  \
0             1  ac233f7e2cb51e79bcd756a6ed27511e   
1             1  2148a42f918d60a38c24066d9e46d2c6   
2             1  815d756a15f03c2adffed505948d80a2   
3             1  812c16799d89c56b11b5e08902565985   
4             1  2997449e348b2a4bf963e82c8873f2f8   
...         ...                               ...   
5338          0  cbfeb1eb820c029f4351e8af73816833   
5339          0  925e9103194684c9d065d845b3f4acc2   
5340          0  28b2da8b002f1d6f746f9af2a8740172   
5341          0  b951d3b4e3fc8d6c7a743b9f78e466e1   
5342          0  37f79aa7ebbd308a5cfdb0fbdb514d51   

                                          review_url_pr  \
0     https://www.amazon.de/gp/customer-reviews/R3OG...   
1     https://www.amazon.de/gp/customer-reviews/R1VE...   
2     https://www.amazon.de/gp/customer-reviews/R2UP...   
3     https://www.amazon.de/gp/customer-reviews/R1E7...   
4     https://www.amazon.de/gp/customer-reviews/R2O4...   
...                                                 ...   
5338  https://www.amazon.de/gp/customer-reviews/RBN7...   
5339  https://www.amazon.de/gp/customer-reviews/RDX3...   
5340  https://www.amazon.de/gp/customer-reviews/RT7M...   
5341  https://www.amazon.de/gp/customer-reviews/R36A...   
5342  https://www.amazon.de/gp/customer-reviews/R18V...   

      manufacturer_comment_pr manufacturer_comment_text_pr  \
0                           0                          NaN   
1                           0                          NaN   
2                           0                          NaN   
3                           0                          NaN   
4                           0                          NaN   
...                       ...                          ...   
5338                        0                          NaN   
5339                        0                          NaN   
5340                        0                          NaN   
5341                        0                          NaN   
5342                        0                          NaN   

      manufacturer_comment_date_pr  has_image_pr  \
0                              NaN             0   
1                              NaN             0   
2                              NaN             0   
3                              NaN             0   
4                              NaN             0   
...                            ...           ...   
5338          

In [18]:
df_ratings1=pd.read_excel(r'C:/Users/Ankita/OneDrive - Infiniti Research/Desktop/Johnson&Johnson/Data/Sales_Ratings_Reviews_neutro.xlsx', sheet_name='unique_col_ratings')
df_ratings1 

date        asin match_type_pr  cumulative_star_rating_pr  \
0     44358  B000W6PY1W         EXACT                        4.4   
1     44360  B000W6PY1W         EXACT                        4.4   
2     44361  B000W6PY1W         EXACT                        4.4   
3     44363  B000W6PY1W         EXACT                        4.4   
4     44364  B000W6PY1W         EXACT                        4.4   
...     ...         ...           ...                        ...   
9193  44745  B09GYRXP9P         EXACT                        4.4   
9194  44746  B09GYRXP9P         EXACT                        4.3   
9195  44749  B09GYRXP9P         EXACT                        4.4   
9196  44753  B09GYRXP9P         EXACT                        4.4   
9197  44762  B09GYRXP9P         EXACT                        4.4   

      cumulative_reviews_pr  cumulative_5_star_reviews_pr  \
0                       907                         599.0   
1                       910                         601.0   
2                       915                         604.0   
3                       919                         606.0   
4                       921                         607.0   
...                     ...                           ...   
9193                     49                          30.0   
9194                     51                          29.0   
9195                     52                          30.0   
9196                     54                          30.0   
9197                     55                          32.0   

      cumulative_4_star_reviews_pr  cumulative_3_star_reviews_pr  \
0                            154.0                         100.0   
1                            155.0                         100.0   
2                            156.0                         101.0   
3                            156.0                         101.0   
4                            157.0                         101.0   
...                            ...                           ...   
9193                          15.0                           2.0   
9194                          14.0                           3.0   
9195                          14.0                           3.0   
9196                          16.0                           3.0   
9197                          16.0                           2.0   

      cumulative_2_star_reviews_pr  cumulative_1_star_reviews_pr  \
0                             27.0                          27.0   
1                             27.0                          27.0   
2                             27.0                          27.0   
3                             28.0                          28.0   
4                             28.0                          28.0   
...                            ...                           ...   
9193                           1.0                           1.0   
9194                           3.0                           2.0   
9195                           3.0                           2.0   
9196                           3.0                           2.0   
9197                           3.0                           2.0   

                                          url_pr         date&asin  
0     https://www.amazon.de/dp/B000W6PY1W/?psc=1  44358_B000W6PY1W  
1     https://www.amazon.de/dp/B000W6PY1W/?psc=1  44360_B000W6PY1W  
2     https://www.amazon.de/dp/B000W6PY1W/?psc=1  44361_B000W6PY1W  
3     https://www.amazon.de/dp/B000W6PY1W/?psc=1  44363_B000W6PY1W  
4     https://www.amazon.de/dp/B000W6PY1W/?psc=1  44364_B000W6PY1W  
...                                          ...               ...  
9193  https://www.amazon.de/dp/B09GYRXP9P/?psc=1  44745_B09GYRXP9P  
9194  https://www.amazon.de/dp/B09GYRXP9P/?psc=1  44746_B09GYRXP9P  
9195  https://www.amazon.de/dp/B09GYRXP9P/?psc=1  44749_B09GYRXP9P  
9196  https://www.amazon.de/dp/B09GYRXP9P/?psc=1  44753_B09GYRXP9P  
9197  https://www.amazon.de/dp/B09GYRXP9P/?psc=1  44762_B09GYRXP

In [19]:
df_placement1=pd.read_excel(r'C:/Users/Ankita/OneDrive - Infiniti Research/Desktop/Johnson&Johnson/Data/Placement_neutro.xlsx',sheet_name='uncommon_col_data')
df_placement1  

date        asin              keyword_pr  page_placement_pr  \
0       44481  B08HT315T6              neutrogena                  1   
1       44481  B08HT315T6              neutrogena                  1   
2       44481  B08HT315T6                    bebe                  3   
3       44481  B08HT315T6   listerine mundspülung                  2   
4       44481  B08HT315T6    neutrogena handcreme                  1   
...       ...         ...                     ...                ...   
167237  44620  B07CDMHVBZ   neutrogena bodylotion                  2   
167238  44620  B07CDMHVBZ  neutrogena anti pickel                  2   
167239  44620  B07CDMHVBZ  neutrogena hydro boost                  2   
167240  44620  B07CDMHVBZ   neutrogena augencreme                  1   
167241  44621  B07CDMHVBZ              neutrogena                  2   

        actual_rank_pr  organic_rank_pr  sponsored_rank_pr  sponsored_pr  \
0                    2              NaN                2.0          True   
1                    5              1.0                NaN         False   
2                  123              NaN               27.0          True   
3                   63              NaN               15.0          True   
4                   41             32.0                NaN         False   
...                ...              ...                ...           ...   
167237              68             52.0                NaN         False   
167238              82             66.0                NaN         False   
167239              75             59.0                NaN         False   
167240              40             32.0                NaN         False   
167241              91             72.0                NaN         False   

        third_party_only_pr  amazon_choice_pr  amazon_bestseller_pr  \
0                      True             False                 False   
1                      True              True                  True   
2                      True             False                 False   
3                      True             False                 False   
4                      True             False                 False   
...                     ...               ...                   ...   
167237                False             False                 False   
167238                False             False                 False   
167239                False             False                 False   
167240                False             False                 False   
167241                False             False                 False   

        prime_pantry_pr  amazon_subscribe_and_save_pr  amazon_prime_pr  \
0                 False                         False            False   
1                 False                          True             True   
2                 False                         False            False   
3                 False                         False            False   
4                 False                         False            False   
...                 ...                           ...              ...   
167237            False                         False            False   
167238            False                         False            False   
167239            False                         False            False   
167240            False                         False            False   
167241            False                         False            False   

        amazon_addon_item_pr  review_count_pr  number_of_images_pr  \
0                      False             62.0                    8   
1                       True             62.0                    8   
2                      False             62.0                    8   
3                      False             62.0                    8   
4                      False             62.0                    8   
...                      ...              ...                  ...   

In [20]:
df_content1=pd.read_excel(r'C:/Users/Ankita/OneDrive - Infiniti Research/Desktop/Johnson&Johnson/Data/Content_neutro.xlsx',sheet_name='Uncommon_col')
df_content1  

date        asin                     content_field_pr condition_pr  \
0      44579  B000W6PY1W                  A+ Content (Exists)            Y   
1      44579  B000W6PY1W                   A+ Images (Exists)            Y   
2      44349  B000W6PY1W         Number of Images (Benchmark)            Y   
3      44282  B000W6PY1W                  A+ Content (Exists)          NaN   
4      44282  B000W6PY1W                  A+ Content (Exists)          NaN   
...      ...         ...                                  ...          ...   
83441  44632  B09NS8Q51H  Length of Product Title (Benchmark)            N   
83442  44632  B09NS8Q51H         Number of Images (Benchmark)            Y   
83443  44632  B09NS8Q51H             Number of Images (Count)            Y   
83444  44632  B09NS8Q51H      Product Description (Benchmark)            Y   
83445  44632  B09NS8Q51H           Product Description Exists            Y   

       compliance_manually_changed_pr      expected_content_pr  \
0                                 0.0                   Exists   
1                                 0.0                        1   
2                                 0.0                   Exists   
3                                 NaN                      NaN   
4                                 0.0                      NaN   
...                               ...                      ...   
83441                             0.0  30 ≤ x ≤ 150 Characters   
83442                             0.0                        6   
83443                             0.0                   Exists   
83444                             0.0                     =>80   
83445                             0.0                   Exists   

                                       actual_content_pr  critical_pr  \
0                                                    Yes        False   
1                                                      5        False   
2                                                     11        False   
3                                                    Yes        False   
4                                                    Yes        False   
...                                                  ...          ...   
83441  194: Neutrogena Anti-Pickel+ Tägliches Waschge...         True   
83442                                                  7        False   
83443                                                  7        False   
83444  173: Das Anti-Pickel+ Tägliches Waschgel von N...         True   
83445  Das Anti-Pickel+ Tägliches Waschgel von Neutro...         True   

       customer_product_pr         date&asin  
0                     True  44579_B000W6PY1W  
1                     True  44579_B000W6PY1W  
2                     True  44349_B000W6PY1W  
3                    False  44282_B000W6PY1W  
4                    False  44282_B000W6PY1W  
...                    ...               ...  
83441                 True  44632_B09NS8Q51H  
83442                 True  44632_B09NS8Q51H  
83443                 True  44632_B09NS8Q51H  
83444                 True  44632_B09NS8Q51H  
83445                 True  44632_B09NS8Q51H  

[83446 rows x 10 columns]

In [21]:
df_reviews1=df_reviews1.drop('date&asin',axis=1)
df_ratings1=df_ratings1.drop('date&asin',axis=1)
df_placement1=df_placement1.drop('date&asin',axis=1)
df_content1=df_content1.drop('date&asin',axis=1)

# Merging Datasets for uncommon columns in the filtered dataset

In [131]:
df_mapped=pd.merge(sales,df_reviews1,on=['date','asin'])

In [132]:
df_mapped

year   date  weeknum  fiscal_year  fiscal_month   retailer  country  \
0     2021  44304       16         2021             4  Amazon 1P  Germany   
1     2021  44304       16         2021             4  Amazon 1P  Germany   
2     2021  44304       16         2021             4  Amazon 1P  Germany   
3     2021  44304       16         2021             4  Amazon 1P  Germany   
4     2021  44304       16         2021             4  Amazon 1P  Germany   
...    ...    ...      ...          ...           ...        ...      ...   
4580  2022  44624        9         2022             3  Amazon 1P  Germany   
4581  2022  44624        9         2022             3  Amazon 1P  Germany   
4582  2022  44624        9         2022             3  Amazon 1P  Germany   
4583  2022  44624        9         2022             3  Amazon 1P  Germany   
4584  2022  44624        9         2022             3  Amazon 1P  Germany   

           manufacturer       category       brand  ... star_rating_pr  \
0     JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...              5   
1     JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...              5   
2     JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...              3   
3     JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...              4   
4     JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...              4   
...                 ...            ...         ...  ...            ...   
4580  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...              1   
4581  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...              5   
4582  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...              5   
4583  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...              5   
4584  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...              2   

     unique_pr                duplicate_group_pr  \
0            1  6bab02b3da40d972a5f7bdafea1e5b36   
1            1  5a3d4e57cac0ee5b25bad4278a613fa8   
2            1  af8043916a19acf7a7dd31a0035abbb1   
3            1  38bf347aebe244be64901a6e03dd4dc0   
4            1  4173b9b75b65790a1be5279f7f89bbae   
...        ...                               ...   
4580         1  27b6bcd7ff01068de9669d9a350641e2   
4581         1  cea6f9d458df47bb5108b3e5a91e0060   
4582         1  abf0cbed99dd894e8aa9dfedc6e7388a   
4583         1  6c7edcf3701ac12bae7ece54b55d871c   
4584         1  dff756905e68f885a8c512fb8a1e4d35   

                                          review_url_pr  \
0     https://www.amazon.de/gp/customer-reviews/R3FJ...   
1     https://www.amazon.de/gp/customer-reviews/R3SV...   
2     https://www.amazon.de/gp/customer-reviews/RRR1...   
3     https://www.amazon.de/gp/customer-reviews/RFR7...   
4     https://www.amazon.de/gp/customer-reviews/R3C1...   
...                                                 ...   
4580  https://www.amazon.de/gp/customer-reviews/R19D...   
4581  https://www.amazon.de/gp/customer-reviews/R3HT...   
4582  https://www.amazon.de/gp/customer-reviews/R3HT...   
4583  https://www.amazon.de/gp/customer-reviews/R1IE...   
4584  https://www.amazon.de/gp/customer-reviews/R1CG...   

     manufacturer_comment_pr manufacturer_comment_text_pr  \
0                          0                          NaN   
1                          0                          NaN   
2                          0                          NaN   
3                          0                          NaN   
4                          0                          NaN   
...                      ...                          ...   
4580                       0                          NaN   
4581                       0                          NaN   
4582                       0                          NaN   
4583                       0                          NaN   
4584                       0                          NaN   

     manufacturer_comment_date_pr  has_image_pr  \
0                             NaN             0   
1                      

In [133]:
df_mapped=pd.merge(df_mapped,df_ratings1,on=['date','asin'])
df_mapped

year   date  weeknum  fiscal_year  fiscal_month   retailer  country  \
0     2021  44304       16         2021             4  Amazon 1P  Germany   
1     2021  44304       16         2021             4  Amazon 1P  Germany   
2     2021  44304       16         2021             4  Amazon 1P  Germany   
3     2021  44304       16         2021             4  Amazon 1P  Germany   
4     2021  44304       16         2021             4  Amazon 1P  Germany   
...    ...    ...      ...          ...           ...        ...      ...   
3014  2022  44617        8         2022             2  Amazon 1P  Germany   
3015  2022  44617        8         2022             2  Amazon 1P  Germany   
3016  2022  44624        9         2022             3  Amazon 1P  Germany   
3017  2022  44624        9         2022             3  Amazon 1P  Germany   
3018  2022  44624        9         2022             3  Amazon 1P  Germany   

           manufacturer       category       brand  ...           author_pr  \
0     JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...  Tsimaki Eleftheria   
1     JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...      Produkttestmum   
2     JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...             Vipcold   
3     JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...             laforge   
4     JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...          Mandragora   
...                 ...            ...         ...  ...                 ...   
3014  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...              Surabh   
3015  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...              Surabh   
3016  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...                 T.R   
3017  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...                 T.R   
3018  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...              ViGina   

     match_type_pr cumulative_star_rating_pr cumulative_reviews_pr  \
0            EXACT                       4.5                   634   
1            EXACT                       4.5                   634   
2            EXACT                       4.5                   634   
3            EXACT                       4.5                   634   
4            EXACT                       4.5                   634   
...            ...                       ...                   ...   
3014         EXACT                       4.4                  2375   
3015         EXACT                       4.4                  2375   
3016         EXACT                       4.6                  1191   
3017         EXACT                       4.6                  1191   
3018         EXACT                       4.6                  1191   

     cumulative_5_star_reviews_pr cumulative_4_star_reviews_pr  \
0                           437.0                        108.0   
1                           437.0                        108.0   
2                           437.0                        108.0   
3                           437.0                        108.0   
4                           437.0                        108.0   
...                           ...                          ...   
3014                       1543.0                        428.0   
3015                       1543.0                        428.0   
3016                        893.0                        179.0   
3017                        893.0                        179.0   
3018                        893.0                        179.0   

     cumulative_3_star_reviews_pr  cumulative_2_star_reviews_pr  \
0                            63.0                          13.0   
1                            63.0                          13.0   
2                            63.0                          13.0   
3                            63.0                          13.0   
4                            63.0                          13.0   
...                           ...                           ...   
3014                        2

In [134]:
df_mapped=pd.merge(df_mapped,df_placement1,on=['date','asin'])
df_mapped

year   date  weeknum  fiscal_year  fiscal_month   retailer  country  \
0      2021  44304       16         2021             4  Amazon 1P  Germany   
1      2021  44304       16         2021             4  Amazon 1P  Germany   
2      2021  44304       16         2021             4  Amazon 1P  Germany   
3      2021  44304       16         2021             4  Amazon 1P  Germany   
4      2021  44304       16         2021             4  Amazon 1P  Germany   
...     ...    ...      ...          ...           ...        ...      ...   
19258  2022  44624        9         2022             3  Amazon 1P  Germany   
19259  2022  44624        9         2022             3  Amazon 1P  Germany   
19260  2022  44624        9         2022             3  Amazon 1P  Germany   
19261  2022  44624        9         2022             3  Amazon 1P  Germany   
19262  2022  44624        9         2022             3  Amazon 1P  Germany   

            manufacturer       category       brand  ... third_party_only_pr  \
0      JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...                True   
1      JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...                True   
2      JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...                True   
3      JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...                True   
4      JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...                True   
...                  ...            ...         ...  ...                 ...   
19258  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...                True   
19259  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...               False   
19260  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...               False   
19261  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...                True   
19262  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA  ...                True   

      amazon_choice_pr amazon_bestseller_pr prime_pantry_pr  \
0                False                False           False   
1                False                False           False   
2                False                False           False   
3                False                False           False   
4                False                False           False   
...                ...                  ...             ...   
19258            False                False           False   
19259             True                 True            True   
19260             True                 True            True   
19261            False                False           False   
19262            False                False           False   

      amazon_subscribe_and_save_pr amazon_prime_pr amazon_addon_item_pr  \
0                            False           False                False   
1                            False           False                False   
2                            False           False                False   
3                            False           False                False   
4                            False           False                False   
...                            ...             ...                  ...   
19258                        False           False                False   
19259                         True            True                 True   
19260                         True            True                 True   
19261                        False           False                False   
19262                        False           False                False   

       review_count_pr number_of_images_pr        type_pr  
0                120.0                   7  brand_keyword  
1                120.0                   7  brand_keyword  
2                120.0                   7  brand_keyword  
3                120.0                   7  brand_keyword  
4                120.0                   7  brand_keyword  
...                ...                 ...            ...  
19258            154.0         

In [135]:
df_mapped=pd.merge(df_mapped,df_content1,on=['date','asin'])

In [136]:
df_mapped.shape

(99224, 154)

Raw Dataset Consolidated shape= 99224, 154
After removing more than 50% missing values, data shape=99224,144
After removing variables with 0 variance, shape of the data= 99224, 137
After removing multicollinear variables, shape of the cleaned data= 99224,106

Need to further check the multicollinarity in the final cleaned dataset


# Missing value detection for filtered and merged dataset with uncommon columns

In [28]:
miss=df_mapped.isnull().sum()

In [29]:
with pd.option_context("display.max_rows", None):
    display(miss)

year                                                                              0
date                                                                              0
weeknum                                                                           0
fiscal_year                                                                       0
fiscal_month                                                                      0
retailer                                                                          0
country                                                                           0
manufacturer                                                                      0
category                                                                          0
brand                                                                             0
asin                                                                              0
title                                                                       

In [30]:
per_missing1=df_mapped.isnull().sum()/len(df_mapped)

In [31]:
with pd.option_context("display.max_rows", None):
    display(per_missing1)

year                                                                          0.000000
date                                                                          0.000000
weeknum                                                                       0.000000
fiscal_year                                                                   0.000000
fiscal_month                                                                  0.000000
retailer                                                                      0.000000
country                                                                       0.000000
manufacturer                                                                  0.000000
category                                                                      0.000000
brand                                                                         0.000000
asin                                                                          0.000000
title                                      

dropping variables with more than 50% missing values

In [137]:
df_mapped.drop(['manufacturer_comment_date_pr','manufacturer_comment_text_pr','author_artist_vc'],
               axis=1, inplace=True)

In [138]:
df_mapped.drop(['promotion_price_pr','promotion_text_pr','gsr_level_4','gsr_level_3','promo_type_pr'],
               axis=1, inplace=True)

In [139]:
df_mapped.drop(['url_pr','review_url_pr'],axis=1, inplace=True)

In [140]:
df_mapped.shape

(99224, 144)

# Finding and Removing variables with 0 variance

In [52]:
df_describe=df_mapped.describe()

In [53]:
df_describe.to_csv('C:/Users/Ankita/OneDrive - Infiniti Research/Desktop/df_statistics.csv')

From this initial level analysis, observing the mean and std, dropping all variables which have 0 
mean and std.

In [141]:
df_mapped.drop(['preordered_revenue_amount_vc','preordered_units_vc','inventory_sell_through_rate_vc',
               'add_on_item_pr','prime_exclusive_pr','manufacturer_comment_pr',
               'compliance_manually_changed_pr'],axis=1, inplace=True)

In [142]:
df_mapped.shape

(99224, 137)

# Before removing duplicates, finding correlation with conversion rate and checking for multicollinearity

In [58]:
corr=df_mapped.corr()

In [57]:
corr

year      date   weeknum  fiscal_year  fiscal_month  \
year                  1.000000  0.767775 -0.683005     0.986021     -0.661615   
date                  0.767775  1.000000 -0.056547     0.765743     -0.057191   
weeknum              -0.683005 -0.056547  1.000000    -0.663462      0.965573   
fiscal_year           0.986021  0.765743 -0.663462     1.000000     -0.683760   
fiscal_month         -0.661615 -0.057191  0.965573    -0.683760      1.000000   
...                        ...       ...       ...          ...           ...   
amazon_addon_item_pr  0.087837  0.052725 -0.076774     0.090946     -0.082814   
review_count_pr       0.305632  0.373692 -0.050166     0.297159     -0.035034   
number_of_images_pr  -0.058151 -0.094890 -0.016794    -0.060575     -0.013622   
critical_pr           0.048094  0.177717  0.127422     0.047160      0.127773   
customer_product_pr  -0.007464  0.061380  0.082586    -0.005011      0.078765   

                      is_competitor  isinventory_vc  issales_vc  \
year                            NaN       -0.038309         NaN   
date                            NaN       -0.052812         NaN   
weeknum                         NaN       -0.000211         NaN   
fiscal_year                     NaN       -0.038765         NaN   
fiscal_month                    NaN        0.001999         NaN   
...                             ...             ...         ...   
amazon_addon_item_pr            NaN       -0.005287         NaN   
review_count_pr                 NaN       -0.007112         NaN   
number_of_images_pr             NaN       -0.081214         NaN   
critical_pr                     NaN        0.001520         NaN   
customer_product_pr             NaN       -0.003592         NaN   

                      first_party_ordered_sales_vc  \
year                                     -0.001057   
date                                      0.073246   
weeknum                                   0.087099   
fiscal_year                               0.007795   
fiscal_month                              0.071924   
...                                            ...   
amazon_addon_item_pr                      0.078429   
review_count_pr                           0.117920   
number_of_images_pr                      -0.049867   
critical_pr                               0.019113   
customer_product_pr                       0.032808   

                      first_party_ordered_sales_local_vc  ...  \
year                                           -0.000035  ...   
date                                            0.075739  ...   
weeknum                                         0.088346  ...   
fiscal_year                                     0.008730  ...   
fiscal_month                                    0.073285  ...   
...                                                  ...  ...   
amazon_addon_item_pr                            0.078297  ...   
review_count_pr                                 0.118907  ...   
number_of_images_pr                            -0.050025  ...   
critical_pr                                     0.019936  ...   
customer_product_pr                             0.033269  ...   

                      amazon_choice_pr  amazon_bestseller_pr  prime_pantry_pr  \
year                          0.066098              0.113163         0.092109   
date                          0.081641              0.122614         0.105285   
weeknum                      -0.009932             -0.036633        -0.023513   
fiscal_year                   0.072279              0.117979         0.096845   
fiscal_month                 -0.022751             -0.046125        -0.033414   
...                                ...                   ...              ...   
amazon_addon_item_pr          0.680562              0.891445         0.687053   
review_count_pr               0.080389              0.011160        -0.019284   
number_of_images_pr          -0.056913             -0.028399        -0.025600   
critical

In [59]:
corr.to_csv('C:/Users/Ankita/OneDrive - Infiniti Research/Desktop/corr.csv')

In [ ]:
Glance Views is causing high level of multicollinearity, so dropping that

In [60]:
df_mapped.drop(['glance_views_vc'],axis=1, inplace=True)

Checking the correlation again

In [61]:
corr=df_mapped.corr()

In [64]:
#corr.to_csv('C:/Users/Ankita/OneDrive - Infiniti Research/Desktop/cor.csv')
corr

year      date   weeknum  fiscal_year  fiscal_month  \
year                  1.000000  0.767775 -0.683005     0.986021     -0.661615   
date                  0.767775  1.000000 -0.056547     0.765743     -0.057191   
weeknum              -0.683005 -0.056547  1.000000    -0.663462      0.965573   
fiscal_year           0.986021  0.765743 -0.663462     1.000000     -0.683760   
fiscal_month         -0.661615 -0.057191  0.965573    -0.683760      1.000000   
...                        ...       ...       ...          ...           ...   
amazon_addon_item_pr  0.087837  0.052725 -0.076774     0.090946     -0.082814   
review_count_pr       0.305632  0.373692 -0.050166     0.297159     -0.035034   
number_of_images_pr  -0.058151 -0.094890 -0.016794    -0.060575     -0.013622   
critical_pr           0.048094  0.177717  0.127422     0.047160      0.127773   
customer_product_pr  -0.007464  0.061380  0.082586    -0.005011      0.078765   

                      is_competitor  isinventory_vc  issales_vc  \
year                            NaN       -0.038309         NaN   
date                            NaN       -0.052812         NaN   
weeknum                         NaN       -0.000211         NaN   
fiscal_year                     NaN       -0.038765         NaN   
fiscal_month                    NaN        0.001999         NaN   
...                             ...             ...         ...   
amazon_addon_item_pr            NaN       -0.005287         NaN   
review_count_pr                 NaN       -0.007112         NaN   
number_of_images_pr             NaN       -0.081214         NaN   
critical_pr                     NaN        0.001520         NaN   
customer_product_pr             NaN       -0.003592         NaN   

                      first_party_ordered_sales_vc  \
year                                     -0.001057   
date                                      0.073246   
weeknum                                   0.087099   
fiscal_year                               0.007795   
fiscal_month                              0.071924   
...                                            ...   
amazon_addon_item_pr                      0.078429   
review_count_pr                           0.117920   
number_of_images_pr                      -0.049867   
critical_pr                               0.019113   
customer_product_pr                       0.032808   

                      first_party_ordered_sales_local_vc  ...  \
year                                           -0.000035  ...   
date                                            0.075739  ...   
weeknum                                         0.088346  ...   
fiscal_year                                     0.008730  ...   
fiscal_month                                    0.073285  ...   
...                                                  ...  ...   
amazon_addon_item_pr                            0.078297  ...   
review_count_pr                                 0.118907  ...   
number_of_images_pr                            -0.050025  ...   
critical_pr                                     0.019936  ...   
customer_product_pr                             0.033269  ...   

                      amazon_choice_pr  amazon_bestseller_pr  prime_pantry_pr  \
year                          0.066098              0.113163         0.092109   
date                          0.081641              0.122614         0.105285   
weeknum                      -0.009932             -0.036633        -0.023513   
fiscal_year                   0.072279              0.117979         0.096845   
fiscal_month                 -0.022751             -0.046125        -0.033414   
...                                ...                   ...              ...   
amazon_addon_item_pr          0.680562              0.891445         0.687053   
review_count_pr               0.080389              0.011160        -0.019284   
number_of_images_pr          -0.056913             -0.028399        -0.025600   
critical

In [ ]:
num=df_mapped.select_dtypes(exclude='object')
num.shape

In [89]:
df=num.copy()

In [90]:
df.drop('conversion_rate_vc',axis=1,inplace=True)

removing highly multi-collinear variables

In [87]:
# Defining a correlation function to extract features that are highly correlated with each other
def correlation (df, threshold):
    col_corr=set() # set of all names of correlated features
    corr_matrix=df.corr()
    for i in range (len(corr_matrix.columns)):
        for j in range (i):
            if abs(corr_matrix.iloc[i,j])>threshold:
                colname=corr_matrix.columns[i]
                col_corr.add(colname)
    return col_corr

In [91]:
corr_features=correlation(df, 0.7)
len(set(corr_features))

35

In [92]:
corr_features
# All these features are highly correlated in our final list and hence dropping all these features

{'actual_rank_pr',
 'aged_90_plus_days_sellable_amount_trailing_30_day_average_amount_vc',
 'aged_90_plus_days_sellable_units_vc',
 'amazon_addon_item_pr',
 'amazon_prime_pr',
 'amazon_subscribe_and_save_pr',
 'available_units_vc',
 'cumulative_1_star_reviews_pr',
 'cumulative_2_star_reviews_pr',
 'cumulative_3_star_reviews_pr',
 'cumulative_4_star_reviews_pr',
 'cumulative_5_star_reviews_pr',
 'date',
 'ean_vc',
 'first_party_ordered_sales_local_vc',
 'first_party_ordered_units_vc',
 'first_party_shipped_sales_vc',
 'first_party_shipped_units_vc',
 'fiscal_month',
 'fiscal_year',
 'list_price_vc',
 'net_received_units_vc',
 'ordered_subcategory_sales_rank_vc',
 'organic_rank_pr',
 'prime_pantry_pr',
 'regular_price_pr',
 'review_count_pr',
 'sellable_on_hand_amount_trailing_30_day_average_amount_vc',
 'sellable_on_hand_units_vc',
 'shipped_cogs_amount_vc',
 'sponsored_rank_pr',
 'unfilled_customer_ordered_units_vc',
 'unhealthy_units_vc',
 'unsellable_on_hand_amount_trailing_30_day_av

In [95]:
corr_features=num[['actual_rank_pr',
 'aged_90_plus_days_sellable_amount_trailing_30_day_average_amount_vc',
 'aged_90_plus_days_sellable_units_vc',
 'amazon_addon_item_pr',
 'amazon_prime_pr',
 'amazon_subscribe_and_save_pr',
 'available_units_vc',
 'cumulative_1_star_reviews_pr',
 'cumulative_2_star_reviews_pr',
 'cumulative_3_star_reviews_pr',
 'cumulative_4_star_reviews_pr',
 'cumulative_5_star_reviews_pr',
 'date',
 'ean_vc',
 'first_party_ordered_sales_local_vc',
 'first_party_ordered_units_vc',
 'first_party_shipped_sales_vc',
 'first_party_shipped_units_vc',
 'fiscal_month',
 'fiscal_year',
 'list_price_vc',
 'net_received_units_vc',
 'ordered_subcategory_sales_rank_vc',
 'organic_rank_pr',
 'prime_pantry_pr',
 'regular_price_pr',
 'review_count_pr',
 'sellable_on_hand_amount_trailing_30_day_average_amount_vc',
 'sellable_on_hand_units_vc',
 'shipped_cogs_amount_vc',
 'sponsored_rank_pr',
 'unfilled_customer_ordered_units_vc',
 'unhealthy_units_vc',
 'unsellable_on_hand_amount_trailing_30_day_average_amount_vc',
 'unsellable_on_hand_units_vc','conversion_rate_vc']]

Checking the correlation of multicollinear features with conversion rate

In [97]:
df=corr_features.corr()

In [98]:
df.to_csv('C:/Users/Ankita/OneDrive - Infiniti Research/Desktop/corr_features.csv')

In [99]:
df_mapped.drop(['actual_rank_pr',
 'aged_90_plus_days_sellable_amount_trailing_30_day_average_amount_vc',
 'aged_90_plus_days_sellable_units_vc',
 'amazon_addon_item_pr',
 'amazon_prime_pr',
 'amazon_subscribe_and_save_pr',
 'available_units_vc',
 'cumulative_1_star_reviews_pr',
 'cumulative_2_star_reviews_pr',
 'cumulative_3_star_reviews_pr',
 'cumulative_4_star_reviews_pr',
 #'cumulative_5_star_reviews_pr',
 'date',
 'ean_vc',
 'first_party_ordered_sales_local_vc',
 #'first_party_ordered_units_vc',
 'first_party_shipped_sales_vc',
 #'first_party_shipped_units_vc',
 'fiscal_month',
 'fiscal_year',
 #'list_price_vc',
 'net_received_units_vc',
 #'ordered_subcategory_sales_rank_vc',
 'organic_rank_pr',
 'prime_pantry_pr',
 'regular_price_pr',
 'review_count_pr',
 'sellable_on_hand_amount_trailing_30_day_average_amount_vc',
 'sellable_on_hand_units_vc',
 'shipped_cogs_amount_vc',
 'sponsored_rank_pr',
 'unfilled_customer_ordered_units_vc',
 'unhealthy_units_vc',
 'unsellable_on_hand_amount_trailing_30_day_average_amount_vc',
 'unsellable_on_hand_units_vc'],axis=1, inplace=True)

In [100]:
df_mapped.shape

(99224, 106)

In [101]:
new_df=df_mapped.corr()

In [102]:
#new_df.to_csv('C:/Users/Ankita/OneDrive - Infiniti Research/Desktop/new_features.csv')

# Segregating categorical and numerical features

In [151]:
df_num=df_mapped.select_dtypes(include=['int64','float64'])

In [152]:
df_num.shape

(99224, 62)

All numerical features which have more than 20% correlation with conversion rate are being considered

In [153]:
final_num_features=df_mapped[[
'first_party_ordered_sales_vc',
'first_party_ordered_sales_local_vc',
'first_party_ordered_units_vc',
'first_party_shipped_sales_vc',
'shipped_cogs_amount_vc',
'first_party_shipped_units_vc',
'conversion_rate_vc',
'unsellable_on_hand_amount_trailing_30_day_average_amount_vc',
'unsellable_on_hand_units_vc',
'unfilled_customer_ordered_units_vc',
'ean_vc',
'cumulative_5_star_reviews_pr',
'number_of_images_pr']]

In [145]:
df_categorical = df_mapped.select_dtypes(include='object')
df_categorical

retailer  country       manufacturer       category       brand  \
0      Amazon 1P  Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA   
1      Amazon 1P  Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA   
2      Amazon 1P  Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA   
3      Amazon 1P  Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA   
4      Amazon 1P  Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA   
...          ...      ...                ...            ...         ...   
99219  Amazon 1P  Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA   
99220  Amazon 1P  Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA   
99221  Amazon 1P  Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA   
99222  Amazon 1P  Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA   
99223  Amazon 1P  Germany  JOHNSON & JOHNSON  EVERYDAY FACE  NEUTROGENA   

             asin                                              title  \
0      B07BVW522M  Neutrogena Hydro Boost Hyaluron Serum, Aqua Pe...   
1      B07BVW522M  Neutrogena Hydro Boost Hyaluron Serum, Aqua Pe...   
2      B07BVW522M  Neutrogena Hydro Boost Hyaluron Serum, Aqua Pe...   
3      B07BVW522M  Neutrogena Hydro Boost Hyaluron Serum, Aqua Pe...   
4      B07BVW522M  Neutrogena Hydro Boost Hyaluron Serum, Aqua Pe...   
...           ...                                                ...   
99219  B0849D9GB4  Neutrogena Hydro Boost Aqua Gel (50ml) - Feuch...   
99220  B0849D9GB4  Neutrogena Hydro Boost Aqua Gel (50ml) - Feuch...   
99221  B0849D9GB4  Neutrogena Hydro Boost Aqua Gel (50ml) - Feuch...   
99222  B0849D9GB4  Neutrogena Hydro Boost Aqua Gel (50ml) - Feuch...   
99223  B0849D9GB4  Neutrogena Hydro Boost Aqua Gel (50ml) - Feuch...   

               gfo gsr_level_1   gsr_level_2  ...  \
0      Skin Health   Face Care  Moisturisers  ...   
1      Skin Health   Face Care  Moisturisers  ...   
2      Skin Health   Face Care  Moisturisers  ...   
3      Skin Health   Face Care  Moisturisers  ...   
4      Skin Health   Face Care  Moisturisers  ...   
...            ...         ...           ...  ...   
99219  Skin Health   Face Care  Moisturisers  ...   
99220  Skin Health   Face Care  Moisturisers  ...   
99221  Skin Health   Face Care  Moisturisers  ...   
99222  Skin Health   Face Care  Moisturisers  ...   
99223  Skin Health   Face Care  Moisturisers  ...   

                     duplicate_group_pr                           ukey_pr  \
0      6bab02b3da40d972a5f7bdafea1e5b36  34a1e61c75b343949210a688cb8e3f87   
1      6bab02b3da40d972a5f7bdafea1e5b36  34a1e61c75b343949210a688cb8e3f87   
2      6bab02b3da40d972a5f7bdafea1e5b36  34a1e61c75b343949210a688cb8e3f87   
3      6bab02b3da40d972a5f7bdafea1e5b36  34a1e61c75b343949210a688cb8e3f87   
4      6bab02b3da40d972a5f7bdafea1e5b36  34a1e61c75b343949210a688cb8e3f87   
...                                 ...                               ...   
99219  6c7edcf3701ac12bae7ece54b55d871c  e086f83236b08e3b5b9c401756f768a9   
99220  6c7edcf3701ac12bae7ece54b55d871c  e086f83236b08e3b5b9c401756f768a9   
99221  6c7edcf3701ac12bae7ece54b55d871c  e086f83236b08e3b5b9c401756f768a9   
99222  6c7edcf3701ac12bae7ece54b55d871c  e086f83236b08e3b5b9c401756f768a9   
99223  6c7edcf3701ac12bae7ece54b55d871c  e086f83236b08e3b5b9c401756f768a9   

                author_pr match_type_pr                            keyword_pr  \
0      Tsimaki Eleftheria         EXACT                            neutrogena   
1      Tsimaki Eleftheria         EXACT                            neutrogena   
2      Tsimaki Eleftheria         EXACT                  neutrogena handcreme   
3      Tsimaki Eleftheria         EXACT                  neutrogena handcreme   
4      Tsimaki Eleftheria         EXACT                 neutrogena bodylotion   
...                   ...           ...                                   ...   
99219              ViGina         EXACT  neutrogena mizellen-reinigungswasser   
99220              ViGina       

In [122]:
##frequency count for cat variables
for i in df_categorical:
    print(i) 
    print('#'*20)
    print(df_categorical[i].value_counts())

retailer
####################
Amazon 1P    99224
Name: retailer, dtype: int64
country
####################
Germany    99224
Name: country, dtype: int64
manufacturer
####################
JOHNSON & JOHNSON    99224
Name: manufacturer, dtype: int64
category
####################
EVERYDAY FACE    99224
Name: category, dtype: int64
brand
####################
NEUTROGENA    99224
Name: brand, dtype: int64
asin
####################
B082VNKJGW    15316
B082VPVQZS    13431
B0849D9GB4     7348
B087ZG94GD     7157
B000W6PY1W     6876
B087ZG9VTV     6845
B087QX6CMP     5028
B087QX6CML     3652
B082VP1JR6     3455
B087ZFLY3K     3335
B087QX6CMQ     3202
B082VNCDC5     3137
B07BVW522M     3130
B087ZGGRL5     2792
B07ZPDV5NJ     2779
B094GC1G7N     2355
B08HT315T6     2344
B08TJRPMQH     2008
B082VNSG4N     1593
B08TJVD6TW     1219
B071936B11      770
B09GYRXP9P      288
B082VNH94D      286
B09NS8Q51H      276
B082VP2KKF      202
B09NLL2YSF      132
B082VMNCNV      104
B075KJBF9H       84
B07JMXSSYN   

prep_instructions_required_vc
####################
-                        98784
Set creation, Bagging       84
Bagging, Set creation       54
Name: prep_instructions_required_vc, dtype: int64
prep_instructions_vendor_state_vc
####################
Amazon            95175
CPEX Certified     2355
Vendor              770
-                   420
Name: prep_instructions_vendor_state_vc, dtype: int64
brand_code_vc
####################
NEUVP    93803
NEUXP     5119
Name: brand_code_vc, dtype: int64
brand_vc
####################
Neutrogena    98922
Name: brand_vc, dtype: int64
manufacturer_code_vc
####################
JOMO1    98922
Name: manufacturer_code_vc, dtype: int64
parent_manufacturer_code_vc
####################
JOMO1    98922
Name: parent_manufacturer_code_vc, dtype: int64
local_currency
####################
EUR    99224
Name: local_currency, dtype: int64
target_currency
####################
USD    99224
Name: target_currency, dtype: int64
date&asin
####################
44583_B08TJR

Removing variables which have only 1 category in all the records

In [111]:
df_categorical['expected_content_pr'].value_counts()

Exists                     37957
1                          24301
30 ≤ x ≤ 150 Characters    12767
6                           7388
=>80                        7388
=>150                       5379
Exist                       2532
Neutrogena                    24
Name: expected_content_pr, dtype: int64

In [156]:
df_final_cat=df_mapped[['gsr_level_2', 'availability_pr','third_party_seller_pr','product_title_vc','product_group_vc','replenishment_code_vc',
'product_group_vc','colour_vc','prep_instructions_vendor_state_vc','brand_code_vc','summary_pr','text_pr','keyword_pr',
'content_field_pr','condition_pr']]

In [ ]:
Total 15 categorical features and 13 numerical features after feature selection through correlation
and a quick look of the value_counts of categorical features

In [158]:
df_final_mapped=df_mapped[[
'first_party_ordered_sales_vc',
'first_party_ordered_sales_local_vc',
'first_party_ordered_units_vc',
'first_party_shipped_sales_vc',
'shipped_cogs_amount_vc',
'first_party_shipped_units_vc',
'conversion_rate_vc',
'unsellable_on_hand_amount_trailing_30_day_average_amount_vc',
'unsellable_on_hand_units_vc',
'unfilled_customer_ordered_units_vc',
'ean_vc',
'cumulative_5_star_reviews_pr',
'number_of_images_pr','gsr_level_2', 'availability_pr','third_party_seller_pr','product_title_vc','product_group_vc','replenishment_code_vc',
'product_group_vc','colour_vc','prep_instructions_vendor_state_vc','brand_code_vc','summary_pr','text_pr','keyword_pr',
'content_field_pr','condition_pr']]

In [159]:
df_final_mapped.shape

(99224, 28)